src/test/java/systemtests/FindCommandSystemTest.java

```diff
@@ -1,176 +1,192 @@
+        assertSelectedCardUnchanged();
+
+        /* Case: find multiple decks in Anakin, 2 keywords in reversed order -> 2 decks found */
+        command = FindCommand.COMMAND_WORD + " Calculus Bacon";
+        assertCommandSuccess(command, expectedModel);
+        assertSelectedCardUnchanged();
+
+        /* Case: find multiple decks in Anakin, 2 keywords with 1 repeat -> 2 decks found */
+        command = FindCommand.COMMAND_WORD + " Calculus Bacon Calculus";
+        assertCommandSuccess(command, expectedModel);
+        assertSelectedCardUnchanged();
+
+        /* Case: find multiple decks in Anakin, 2 matching keywords and 1 non-matching keyword
+         * -> 2 decks found
+         */
+        command = FindCommand.COMMAND_WORD + " Calculus Bacon NonMatchingKeyWord";
+        assertCommandSuccess(command, expectedModel);
+        assertSelectedCardUnchanged();
+
+        /* Case: undo previous find command -> rejected */
+        command = UndoCommand.COMMAND_WORD;
+        String expectedResultMessage = UndoCommand.MESSAGE_FAILURE;
+        assertCommandFailure(command, expectedResultMessage);
+
+        /* Case: redo previous find command -> rejected */
+        command = RedoCommand.COMMAND_WORD;
+        expectedResultMessage = RedoCommand.MESSAGE_FAILURE;
+        assertCommandFailure(command, expectedResultMessage);
+
+        /* Case: find same decks in Anakin after deleting 1 of them -> 1 deck found */
+        executeCommand(DeleteDeckCommand.COMMAND_WORD + " 1");
+        assertFalse(getModel().getAnakin().getDeckList().contains(DECK_G));
+        command = FindCommand.COMMAND_WORD + " " + KEYWORD_MATCHING_JOHN;
+        expectedModel = getModel();
+        ModelHelper.setFilteredDeckList(expectedModel, DECK_H);
+        assertCommandSuccess(command, expectedModel);
+        assertSelectedCardUnchanged();
+
+        /* Case: find deck in Anakin, keyword is same as name but of different case -> 1 person found */
+        command = FindCommand.COMMAND_WORD + " JoHn";
+        assertCommandSuccess(command, expectedModel);
+        assertSelectedCardUnchanged();
+
+        /* Case: find person in Anakin, keyword is substring of name -> 0 persons found */
+        command = FindCommand.COMMAND_WORD + " Joh";
+        ModelHelper.setFilteredDeckList(expectedModel);
+        assertCommandSuccess(command, expectedModel);
+        assertSelectedCardUnchanged();
+
+        /* Case: find person in Anakin, name is substring of keyword -> 0 persons found */
+        command = FindCommand.COMMAND_WORD + " Johnny";
+        ModelHelper.setFilteredDeckList(expectedModel);
+        assertCommandSuccess(command, expectedModel);
+        assertSelectedCardUnchanged();
+
+        /* Case: find person not in Anakin -> 0 persons found */
+        command = FindCommand.COMMAND_WORD + " Julius";
+        assertCommandSuccess(command, expectedModel);
+        assertSelectedCardUnchanged();
+        // TODO: Write Card level checks for find
```

Comment:
Add Card level checks for find

Reply:
This will be resolved in a follow up PR on card level commands for all operations